src/main.jakt

```diff
@@ -25,101 +25,111 @@ function main(args: [String]) {
             mut cpu = CPU::init(system)
             mut debugger = Debugger::init(cpu)
             mut joypad = system.joypad
-	        mut ppu = system.ppu
-            loop {
-                let in_vblank_before = system.ppu.in_vblank
-
-                let cpu_clocks_before = cpu.clock
-                cpu.run_opcode()
-                let cpu_clocks_after = cpu.clock
-
-                if system.debug {
-                    debugger.dump_cpu_state()
-                }
+            mut ppu = system.ppu
+            mut clock_pause: bool = false
+            mut one_frame: bool = false
+            mut slow_mode: bool = false
+            eprintln("Debugging keys\n--------------")
+            eprintln("D - hold to dump current CPU info\nH - toggle CPU halt\nJ - When halted, run single clock\nL - toggle slow mode")
 
-//                mut ppu = system.ppu
-                ppu.tick(cycles: (cpu_clocks_after - cpu_clocks_before) * 3)
-                let in_vblank_after = system.ppu.in_vblank
-
-                if not in_vblank_before and in_vblank_after {
-                    // We're just now in the vblank
-                    draw_frame(sdl, buffer: ppu.video_buffer)
-                    mut event = sdl.poll_event()
-                    if event.has_value() {
-                        match event! {
-                            Quit => {
-                                break
-                            }
-                            KeyDown(key) => {
-                                match key {
-                                    Up => {
-                                        joypad.press_up()
-                                    }
-                                    Down => {
-                                        joypad.press_down()
-                                    }
-                                    Left => {
-                                        joypad.press_left()
-                                    }
-                                    Right => {
-                                        joypad.press_right()
-                                    }
-                                    Return => {
-                                        joypad.press_start()
-                                    }
-                                    Tab => {
-                                        joypad.press_select()
-                                    }
-                                    Z => {
-                                        joypad.press_a()
-                                    }
-                                    X => {
-                                        joypad.press_b()
-                                    }
-                                    D => {
-                                        system.debug_enable()
-                                    }
+            loop {
+                mut event = sdl.poll_event()
```

Comment:
Oh, this might be it. I'll do some research but we shouldn't check for events every pixel, only every frame.

Reply:
Yeah that seems to be it. Working on a fix